### DISCUSSION TITLE
How do I specify multiple tls certificates to be used for different hostnames in a bun server?

### DISCUSSION THREAD

**Discussion by coffeebite** [oven-sh/bun#8916] [Q&A] [answered]
I can configure TLS as follows:

```js
const server = Bun.serve({
  fetch: (request) => new Response("Welcome to Bun!"),
  tls: {
    cert: Bun.file("cert.pem"),
    key: Bun.file("key.pem"),
  },
});
```

My server needs multiple certificates, each to be used for a different hostname. Is there a way to do so?

**Comment by truongan07** [comment]
You need to implement the microservice architecture where each service has one or more Bun's server inside this service. Ex : navigate server, API server, websocket server, webhook server, ... and they are unique. And then, you implement multiple services that you have an ecosystem with multiple certificates, each to be used for a different hostname

**Reply by coffeebite** [reply]
I have the same service that listens on multiple hostnames. Deploying it as two separate services works but I was wondering if there was a way to avoid that.

**Comment by truongan07** [comment, marked as answer]
> I have the same service that listens on multiple hostnames. Deploying it as two separate services works but I was wondering if there was a way to avoid that.

[Bun.serve() now supports Server Name Indication (SNI)](https://bun.sh/blog/bun-v1.1.5#bun-serve-now-supports-server-name-indication-sni)

**Comment by coffeebite** [comment]
Thank you @truongan07.
